modules/utxo-lib/src/bitgo/zcash/ZcashTransaction.ts

```diff
@@ -0,0 +1,390 @@
+        /* istanbul ignore next */
+        throw new Error(`shielded spend not supported`);
+      }
+
+      const nShieldedOutput = bufferReader.readVarInt();
+      if (nShieldedOutput !== 0) {
+        /* istanbul ignore next */
+        throw new Error(`shielded output not supported`);
+      }
+    }
+
+    if (tx.supportsJoinSplits()) {
+      const joinSplitsLen = bufferReader.readVarInt();
+      if (joinSplitsLen !== 0) {
+        /* istanbul ignore next */
+        throw new Error(`joinSplits not supported`);
+      }
+    }
+
+    if (__noStrict) return tx;
+    if (bufferReader.offset !== buffer.length) throw new Error('Transaction has unexpected data');
+
+    return tx;
+  }
+
+  byteLength() {
+    let byteLength = super.byteLength();
+    if (this.isOverwinterCompatible()) {
+      byteLength += 4; // nVersionGroupId
+    }
+    if (this.isOverwinterCompatible()) {
+      byteLength += 4; // nExpiryHeight
+    }
+    if (this.isSaplingCompatible()) {
+      byteLength += 8; // valueBalance
+      byteLength += varuint.encodingLength(0); // inputs
+      byteLength += varuint.encodingLength(0); // outputs
+    }
+    if (this.supportsJoinSplits()) {
+      byteLength += varuint.encodingLength(0); // joinsplits
+    }
+    return byteLength;
+  }
+
+  isSaplingCompatible() {
+    return this.overwintered && this.version >= ZcashTransaction.VERSION_SAPLING;
+  }
+
+  isOverwinterCompatible() {
+    return this.overwintered && this.version >= ZcashTransaction.VERSION_OVERWINTER;
+  }
+
+  supportsJoinSplits() {
+    return this.overwintered && this.version >= ZcashTransaction.VERSION_JOINSPLITS_SUPPORT;
+  }
+
+  /**
+   * Build a hash for all or none of the transaction inputs depending on the hashtype
+   * @param hashType
+   * @returns Buffer - BLAKE2b hash or 256-bit zero if doesn't apply
+   */
+  getPrevoutHash(hashType) {
+    if (!(hashType & Transaction.SIGHASH_ANYONECANPAY)) {
+      const bufferWriter = new BufferWriter(Buffer.allocUnsafe(36 * this.ins.length));
+
+      this.ins.forEach(function (txIn) {
+        bufferWriter.writeSlice(txIn.hash);
+        bufferWriter.writeUInt32(txIn.index);
+      });
+
+      return getBlake2bHash(bufferWriter.buffer, 'ZcashPrevoutHash');
+    }
+    return ZERO;
+  }
+
+  /**
+   * Build a hash for all or none of the transactions inputs sequence numbers depending on the hashtype
+   * @param hashType
+   * @returns Buffer BLAKE2b hash or 256-bit zero if doesn't apply
+   */
+  getSequenceHash(hashType) {
+    if (
+      !(hashType & Transaction.SIGHASH_ANYONECANPAY) &&
+      (hashType & 0x1f) !== Transaction.SIGHASH_SINGLE &&
+      (hashType & 0x1f) !== Transaction.SIGHASH_NONE
+    ) {
+      const bufferWriter = new BufferWriter(Buffer.allocUnsafe(4 * this.ins.length));
+
+      this.ins.forEach(function (txIn) {
+        bufferWriter.writeUInt32(txIn.sequence);
+      });
+
+      return getBlake2bHash(bufferWriter.buffer, 'ZcashSequencHash');
```

Comment:
Is this a typo?
```suggestion
      return getBlake2bHash(bufferWriter.buffer, 'ZcashSequenceHash');
```

Reply:
I guess not: https://github.com/BitGo/BitGoJS/blob/master/modules/utxo-lib/src/transaction.js#L583

Reply:
I think it's a character limit